bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/base/BlackBodySpectrum.java

```diff
@@ -75,23 +74,18 @@ private double _convertToMag(final double flux, final BrightnessUnit units, fina
         // into a genral purpose conversion class if needs to be used again.
         final double norm;
         //The firstpart converts the units to our internal units.
-        switch (units) {
-            case JY:                norm = flux * 1.509e7 / band.center().toNanometers();       break;
-            case WATTS:             norm = flux * band.center().toNanometers() / 1.988e-13;     break;
-            case ERGS_WAVELENGTH:   norm = flux * band.center().toNanometers() / 1.988e-14;     break;
-            case ERGS_FREQUENCY:    norm = flux * 1.509e30 / band.center().toNanometers();      break;
-            case ABMAG:             norm = 5.632e10 * Math.pow(10, -0.4 * flux) / band.center().toNanometers(); break;
-            case MAG_PSA:
-                double zeropoint = ZeroMagnitudeStar.getAverageFlux(band);
-                norm = zeropoint * (Math.pow(10.0, -0.4 * flux));
-                break;
-            case JY_PSA:            norm = flux * 1.509e7 / band.center().toNanometers();       break;
-            case WATTS_PSA:         norm = flux * band.center().toNanometers() / 1.988e-13;     break;
-            case ERGS_WAVELENGTH_PSA: norm = flux * band.center().toNanometers() / 1.988e-14;   break;
-            case ERGS_FREQUENCY_PSA:norm = flux * 1.509e30 / band.center().toNanometers();      break;
-            case ABMAG_PSA:         norm = 5.632e10 * Math.pow(10, -0.4 * flux) / band.center().toNanometers(); break;
-            default:
-                throw new IllegalArgumentException("invalid units " + units);
+        if (units.equals(MagnitudeSystem.AB$.MODULE$) || units.equals(SurfaceBrightness.AB$.MODULE$)) {
+            norm = 5.632e10 * Math.pow(10, -0.4 * flux) / band.center().toNanometers();
+        } else if (units.equals(MagnitudeSystem.Jy$.MODULE$) || units.equals(SurfaceBrightness.Jy$.MODULE$)) {
+            norm = flux * 1.509e7 / band.center().toNanometers();
+        } else if (units.equals(MagnitudeSystem.Watts$.MODULE$) || units.equals(SurfaceBrightness.Watts$.MODULE$)) {
+            norm = flux * band.center().toNanometers() / 1.988e-13;
+        } else if (units.equals(MagnitudeSystem.ErgsWavelength$.MODULE$) || units.equals(SurfaceBrightness.ErgsWavelength$.MODULE$)) {
+            norm = flux * band.center().toNanometers() / 1.988e-14;
+        } else if (units.equals(MagnitudeSystem.ErgsFrequency$.MODULE$) || units.equals(SurfaceBrightness.ErgsFrequency$.MODULE$)) {
+            norm = flux * 1.509e30 / band.center().toNanometers();
+        } else {
+            throw new IllegalArgumentException("invalid units " + units);
```

Comment:
I think this would work better if it were written in Scala … this seems hard to maintain and doesn't provide exhaustiveness checking.